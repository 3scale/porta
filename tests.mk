#test-prep:
#	$(MAKE) init_db_with_deps test-with-info CMD="#{CMD}"
#test-prep: bundle npm-install test-with-info

ifeq ($(PROXY_ENABLED),true)
rake_wrapper: CMD = make dnsmasq_set; bundle exec rake $${JOB} --verbose --trace; export TEST_EXIT_CODE=$$? && echo 'tests exit code: $$TEST_EXIT_CODE'; make dnsmasq_unset; exit $$TEST_EXIT_CODE
else
rake_wrapper: CMD = bundle exec rake $${JOB} --verbose --trace
endif
rake_wrapper:
	$(MAKE) test-with-info CMD="${CMD}"

test-rake:
	$(MAKE) bundle npm-install rake_wrapper JOB="${JOB}"

test-cucumber: JOB = integrate:cucumber
test-cucumber: PROXY_ENABLED = true
test-cucumber:
	$(MAKE) bundle npm-install test-rake JOB="${JOB}"

test-lint: bundle
	$(MAKE) test-rake JOB="ci:license_finder:run doc:swagger:validate:all doc:swagger:generate:all ci:jspm integrate:frontend"

