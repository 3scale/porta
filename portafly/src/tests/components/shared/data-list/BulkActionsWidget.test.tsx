import React from 'react'

import { render } from 'tests/custom-render'
import { fireEvent } from '@testing-library/react'
import {
  BulkActionsWidget,
  useDataListTable,
  useDataListBulkActions,
  BulkAction
} from 'components'

jest.mock('components/shared/data-list/DataListContext')

const actions = {
  sendEmail: 'Send email',
  changeState: 'Change status'
}

const setup = (selectedRows: Array<any> = []) => {
  (useDataListTable as jest.Mock).mockReturnValue({ selectedRows });
  (useDataListBulkActions as jest.Mock).mockReturnValue({})
  return render(<BulkActionsWidget actions={actions} />)
}

it('should render properly when closed', () => {
  const wrapper = setup()
  expect(wrapper.container.firstChild).toMatchSnapshot()
})

it('expands and collapses properly', () => {
  const wrapper = setup()
  expect(wrapper.queryByRole('menu')).not.toBeInTheDocument()

  const button = wrapper.getByRole('button')
  fireEvent.click(button)
  expect(wrapper.queryByRole('menu')).toBeInTheDocument()

  fireEvent.click(button)
  expect(wrapper.queryByRole('menu')).not.toBeInTheDocument()
})

describe('when it is disabled', () => {
  it('shows a warning inside the dropdown when opened', () => {
    const wrapper = setup()
    const button = wrapper.getByRole('button')
    fireEvent.click(button)
    expect(wrapper.getByText('bulk_actions.warning')).toBeInTheDocument()
  })
})

describe('when it is enabled', () => {
  it('shows the list of options when opened', () => {
    const wrapper = setup(['element'])
    const button = wrapper.getByRole('button')
    fireEvent.click(button)

    Object.keys(actions).forEach((key) => {
      expect(wrapper.getByText(actions[key as NonNullable<BulkAction>])).toBeInTheDocument()
    })
  })
})
