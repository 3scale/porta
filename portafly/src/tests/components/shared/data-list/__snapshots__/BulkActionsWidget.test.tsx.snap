// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly when closed 1`] = `
<div
  class="pf-c-dropdown"
  data-ouia-component-id="0"
  data-ouia-component-type="PF4/Dropdown"
  data-ouia-safe="true"
  id="data-list-bulk-actions-dropdown"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="pf-c-dropdown__toggle pf-m-primary"
    id="pf-toggle-id-0"
    type="button"
  >
    <span
      class="pf-c-dropdown__toggle-text"
    >
      bulk_actions.title
    </span>
    <span
      class="pf-c-dropdown__toggle-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 320 512"
        width="1em"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          transform=""
        />
      </svg>
    </span>
  </button>
</div>
`;
